src/rrdcached.c

```diff
@@ -74,8 +74,9 @@ static int value_list_to_string(char *buffer, int buffer_len,
 
   memset(buffer, '\0', buffer_len);
 
-  t = CDTIME_T_TO_TIME_T(vl->time);
-  status = snprintf(buffer, buffer_len, "%lu", (unsigned long)t);
+  t = CDTIME_T_TO_DOUBLE(vl->time);
```

Comment:
`t` has the wrong type now (should be `double`, is `time_t`. I'd recommend to remove `t` completely and inline the conversion:

```c
status = snprintf(buffer, buffer_len, "%.6f", CDTIME_T_TO_DOUBLE(vl->time));
```